src/v/cluster/controller.cc

```diff
@@ -612,6 +615,50 @@ controller::create_cluster(const bootstrap_cluster_cmd_data cmd_data) {
     }
 }
 
+ss::future<>
+controller::set_initial_members(std::vector<model::broker> initial_brokers) {
+    vassert(
+      ss::this_shard_id() == controller_stm_shard,
+      "Cluster can only be created from controller_stm_shard");
+    if (initial_brokers.empty()) {
+        co_return;
+    }
+
+    for (simple_time_jitter<model::timeout_clock> retry_jitter(1s);;) {
+        while (_members_table.local().nodes().empty() && !_raft0->is_leader()) {
+            co_await ss::sleep_abortable(100ms, _as.local());
```

Comment:
maybe sleep_abortable should be typedef'ed to be `vsleep()` like `vassert` so we can always abort it.